developers/contracts/xlink-staking.md

```diff
@@ -0,0 +1,458 @@
+    sm1 -.-o box1[["**tx-sender**: User
+    **contract-caller**: DAO extension"]]
+    sm2 -.-o box2[["**tx-sender** & **contract-caller**: DAO extension"]]
+```
+
+The `as-contract` call type is the most common approach for liquid staking implementations, where the intermediary contract stakes on behalf of users and provides them with a receipt in the form of a new (rebasing) token.
+
+#### Governance call
+
+```mermaid
+flowchart LR
+    Operator --Sends transtaction--> dao["Executor DAO"]
+    dao --as-contract--> i["Proposal 
+    contract"]
+    i --contract-call--> sm[Staking Manager]
+    sm -.-o box[["**tx-sender**: Executor DAO 
+    **contract-caller**: Proposal contract"]]
+```
+
+### Mechanics
+
+In the Staking Manager, the staker is represented by the `tx-sender`. As explained above, depending on the intermediary contract design, the staker may either be an end user or the endpoint/façade acting on behalf of the user. Therefore, every reference to "staker" (or `user`, as defined within the contract) applies to any of these two possibilities.
+
+Upon staking, the shares corresponding to the amount being staked are calculated and stored in the [`user-shares`](#user-shares) map. This is the staking position and represents the user's portion of the total amount staked. During this operation, the amount to stake is transferred from the `tx-sender` to the Staking Manager, and since the total amount staked and total shares increased, the contract storage is updated to reflect the new state. Refer to the [`stake`](#stake) function for detailed information.
+
+Unstaking performs the inverse operations. It first calculates the shares that correspond to the amount willing to unstake. Then, the unstaked amount is transferred from the Staking Manager to the `tx-sender` and, finally, state upates are performed to decrease user shares, total shares and total amount staked.
+
+{% hint style="info" %}
+Note that both **staking** and **unstaking** operations involve shares management and mutate the staking status of a token in three dimensions: total shares, total amount staked and user's share-based staking position.
+{% endhint %}
+
+#### Amount to shares conversion
+
+On every staking and unstaking action, an amount-to-shares conversion is perfomed using the following equation:
+
+$$
+\textrm{Shares} = \frac{\textrm{Amount}}{\textrm{Total Staked}} \; \cdot \; \textrm{Total Shares}.
+$$
+
+This is how [`get-shares-given-amount`](#supporting-features) function works. The ratio between the total staked amount and the total shares determines the "share price" in token units, i.e., how many tokens one share represents:
+
+$$
+\textrm{Price} = \frac{\textrm{Total Staked}}{\textrm{Total Shares}}
+$$
+
+Before completing a staking or unstaking operation, these values are updated in a way that the share price remains constant. In contrast, reward accrual operations modify this price by maintaing total shares constant and incresing the total staked value.
```

Comment:
Yes. Perhaps it would be interesting to mention how is it done (that the price remains constant).. as it should be simple: the price formula above shows the price is the ratio of amount-of-shares vs. amount-of-tokens.  Staking adds tokens, and for it to keep the price fixed, we should add shares with the same ratio given by the price, resulting in share minting. Analogously when you want to unstake you'll withdraw your tokens... to conserve the price, shares should be burn with the same ratio tokens vs shares were present before the unstake.

Reply:
^^^  Easy to say ..  not easy to explain/describe...

Reply:
Ok... sounds reasonable. Added a clarification on this.